translator/source_code/icone_translator.py

```diff
@@ -353,6 +364,34 @@ def parse_incident(incident):
 
     return feature
 
+#function to validate the incident
+def validate_incident(incident):
+    
+    if not incident or (type(incident) != dict and type(incident) != OrderedDict):
+        logging.warning('incident is empty or has invalid type')
+        return False
+    try:
+        coords = parse_polyline(incident['location']['polyline'])
+        incident['starttime']
```

Comment:
Rather than relying on this dictionary to thrown an error if the key isn't found (which again is slow), we should check for the existence ourselves:
```
starttime = incident.get('starttime')
```
Then we can check if starttime is None or a value.

Reply:
i did it and pushed the updated code to dev